site/src/sphinx/server-thrift.rst

```diff
@@ -0,0 +1,144 @@
+.. _`Calling a Thrift service`: client-thrift.html
+.. _`ServerBuilder`: apidocs/index.html?com/linecorp/armeria/server/ServerBuilder.html
+.. _`THttpService`: apidocs/index.html?com/linecorp/armeria/server/thrift/THttpService.html
+.. _`TMultiplexedProcessor`: https://github.com/apache/thrift/blob/400b346db2510fffa06c0ced11105e3618ce5367/lib/java/src/org/apache/thrift/TMultiplexedProcessor.java#L28
+
+.. _server-thrift:
+
+Running a Thrift service
+========================
+
+Let's assume we have the following Thrift IDL:
+
+.. code-block:: thrift
+
+    namespace java com.example.thrift.hello
+
+    service HelloService {
+        string hello(1:string name)
+    }
+
+The Apache Thrift compiler will produce some Java code under the ``com.example.thrift.hello`` package.
+The most noteworthy one is ``HelloService.java`` which defines the service interfaces we will implement:
+
+.. code-block:: java
+
+    import org.apache.thrift.TException;
+    import org.apache.thrift.async.AsyncMethodCallback;
+
+    public class HelloService {
+        public interface Iface {
+            public String hello(String name) throws TException;
+        }
+
+        public interface AsyncIface {
+            public void hello(String name, AsyncMethodCallback<String> resultHandler) throws TException;
+        }
+        ...
+    }
+
+If you are interested in going fully asynchronous, it is recommended to implement the ``AsyncIface`` interface,
+while it is slightly easier to implement the synchronous ``Iface`` interface:
```

Comment:
I think it's a lot more than "slightly" easier ;) It would probably be worth noting that with AsyncIface, any blocking code like jdbc calls must be done on a separate threadpool, which is not the case for Iface.